src/main/java/com/streamr/client/utils/EncryptionUtil.java

```diff
@@ -88,17 +139,19 @@ public static void encryptStreamMessageAndNewKey(String newGroupKeyHex, StreamMe
     message content and returns null.
      */
     public static SecretKey decryptStreamMessage(StreamMessage streamMessage, SecretKey groupKey) throws UnableToDecryptException {
+        StreamMessage.EncryptionType encryptionType = streamMessage.getEncryptionType();
         try {
-            if (streamMessage.getEncryptionType() == StreamMessage.EncryptionType.AES) {
+            if (encryptionType == StreamMessage.EncryptionType.AES) {
                 streamMessage.setEncryptionType(StreamMessage.EncryptionType.NONE);
                 streamMessage.setSerializedContent(decrypt(streamMessage.getSerializedContent(), groupKey));
-            } else if (streamMessage.getEncryptionType() == StreamMessage.EncryptionType.NEW_KEY_AND_AES) {
+            } else if (encryptionType == StreamMessage.EncryptionType.NEW_KEY_AND_AES) {
                 byte[] plaintext = EncryptionUtil.decrypt(streamMessage.getSerializedContent(), groupKey);
                 streamMessage.setEncryptionType(StreamMessage.EncryptionType.NONE);
                 streamMessage.setSerializedContent(Arrays.copyOfRange(plaintext, 32, plaintext.length));
```

Comment:
This is caught in the `decrypt` method because the `StreamMessage.setSerializedContent` method will try to parse the "decrypted" content as JSON and fail to do so. This will in turn throw an `UnableToDecryptException` which was previously indeed not gracefully handled. This is now fixed: the error is propagated to a new callback `onUnableToDecrypt` in the `MessageHandler`. By default, that callback just logs the error but the user can override it.